neutrino.go

```diff
@@ -1081,6 +1081,24 @@ func (s *ChainService) BanPeer(addr string, reason banman.Reason) error {
 	return s.banStore.BanIPNet(ipNet, reason, BanDuration)
 }
 
+// UnbanPeer connects and unbans a previously banned peer.
+func (s *ChainService) UnbanPeer(addr string, parmanent bool) error {
+	log.Infof("UnBanning peer %v", addr)
+
+	ipNet, err := banman.ParseIPNet(addr, nil)
+	if err != nil {
+		return fmt.Errorf("unable to parse IP network for peer %v: %v",
+			addr, err)
+	}
+
+	err = s.banStore.UnbanIPNet(ipNet)
+	if err != nil {
+		return fmt.Errorf("unable to unban peer: %v", err)
+	}
+
+	return s.ConnectNode(addr, parmanent)
```

Comment:
sorry for my earlier suggestion, I think this should be in a goroutine so we aren't blocking?

cc @Roasbeef

Reply:
I think the blocking you are referring to might emanate from here?
https://github.com/lightninglabs/neutrino/blob/9130a5e9308d15f2dd9306974d98deffda277b4d/notifications.go#L351-L360
From what I have seen from other exported functions on the chainservice, I do not think they take into account this blocking?

Reply:
See above in `BanPeer`, the disconnect is in a goroutine

Reply:
Thanks for pointing this out. The `Disconnect` in that function is quite different as it is not a chain service method like what I was referring to. I was referring to other [exported chainservice methods](https://github.com/lightninglabs/neutrino/blob/9130a5e9308d15f2dd9306974d98deffda277b4d/notifications.go#L282-L327) that send a message to the query channel and how I have seen that they do not take into account a potential blocking.

I am not sure if we should continue with that practice.

Reply:
It depends on how it's used in practice, which it's not here. So I'm fine with it as-is and we can adjust it later based on usage